src/main/java/com/rackspace/jenkins_nodepool/NodePoolSlave.java

```diff
@@ -181,47 +252,149 @@ public SlaveDescriptor getDescriptor() {
     }
 
     /**
-     * Test if the slave's build is done.
+     * Test if the slave's build is done.  A slave is considered complete when we've executed at least one build and all
+     * the executors are idle, otherwise we are not done.
      *
-     * @return true if the build associated with this slave has completed.
+     * @return true if the build associated with this slave has completed and the executors for the build are idle.
      */
     boolean isBuildComplete() {
 
         final NodePoolComputer computer = (NodePoolComputer) toComputer();
         if (computer == null) {
             return false;
         }
+
         final RunList builds = computer.getBuilds();
-        if (builds == null || !builds.iterator().hasNext()) {
-            // unused
+
+        // DEBUG
+        printBuildDetails(builds);
+        printExecutorDetails(computer.getAllExecutors());
```

Comment:
Do we need these in production? Quite a log of logging for every nodepool node every 60s..

Reply:
Logging is at Level.FINE, but I can remove the calls to the printout.

Reply:
Commented out.